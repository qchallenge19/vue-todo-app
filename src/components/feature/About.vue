<template>
  <app-card v-bind:contents="data.description" v-bind:title="data.title"></app-card>
</template>
<script>
import AppCard from '../shared/AppCard';
export default {
  name: 'About',
  components: {
    'app-card': AppCard
  },
  data: () => {
    return {
      data: {
        title : 'Todo App',
        description: 'The Todo app lets user to add, update and remove the Todo items from the list dynamically. Todo list items persist\n' +
                '    even after refresh of the app. Contents are being served from local JSON file.'
      }
    }
  }
  }
</script>
