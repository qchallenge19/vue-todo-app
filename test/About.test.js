import Vue from "vue";
import About from '../src/components/feature/About';

/* eslint-env jest */
describe('About.vue', () => {
    let cmp, vm;
    const data = {
        title : 'Todo App',
            description: ''
    }
    beforeEach(() => {
        cmp = Vue.extend(About); // Create a copy of the component
        vm = new cmp({
            data: {
                title : 'Todo App',
                description: ''
            }

        }).$mount(); // Instantiates and mounts the component
    });

    test('is a Vue instance', () => {
        expect(vm).toBeTruthy();
    })
    test('should test title of the page', () => {
        expect(vm.title).toEqual(data.title);
    })
    test('should test description of the page', () => {
        expect(vm.description).toEqual(data.description);
    })
})

