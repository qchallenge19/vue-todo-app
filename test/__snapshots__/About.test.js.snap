// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`About.vue renders correctly 1`] = `
<el-row
  gutter="12"
>
  <el-col
    span="24"
  >
    <el-card
      shadow="always"
    >
      <div
        class="desc"
      >
        <div>
          <p>
            Todo App
          </p>
        </div>
         
        <!---->
         
        <div>
          <p>
            The Todo app lets user to add, update and remove the Todo items from the list dynamically. Todo list items persist
    even after refresh of the app. Contents are being served from local JSON file.
          </p>
        </div>
         
        <!---->
      </div>
    </el-card>
  </el-col>
</el-row>
`;
